runtime/vm/BytecodeInterpreter.hpp

```diff
@@ -4021,13 +4021,48 @@ class INTERPRETER_CLASS
 		return EXECUTE_BYTECODE;
 	}
 
-	/* ToDo: unimplemented - https://github.com/eclipse-openj9/openj9/issues/13614 */
 	/* jdk.internal.misc.Unsafe: public native boolean isFlattenedArray(Class<?> clz); */
 	VMINLINE VM_BytecodeAction
 	inlUnsafeIsFlattenedArray(REGISTER_ARGS_LIST)
 	{
+		j9object_t cls = *(j9object_t*)_sp;
+
 		updateVMStruct(REGISTER_ARGS);
-		Assert_VM_unreachable();
+		buildInternalNativeStackFrame(REGISTER_ARGS);
+
+		I_32 result = (I_32)FALSE;
+		if (NULL != cls) {
+			J9Class *fieldClazz = J9VM_J9CLASS_FROM_HEAPCLASS(_currentThread, cls);
+			if (J9CLASS_IS_ARRAY(fieldClazz)) {
+				result = (I_32)J9_IS_J9CLASS_FLATTENED(fieldClazz);
+			}
+		}
+
+		restoreInternalNativeStackFrame(REGISTER_ARGS);
+		returnSingleFromINL(REGISTER_ARGS, result, 2);
+		return EXECUTE_BYTECODE;
+	}
+
+	VMINLINE VM_BytecodeAction
+	inlUnsafeIsFieldFlattenedHelper(REGISTER_ARGS_LIST)
+	{
+		j9object_t field = *(j9object_t*)_sp;
+
+		updateVMStruct(REGISTER_ARGS);
+		buildInternalNativeStackFrame(REGISTER_ARGS);
+
+		J9JNIFieldID *fieldID = _vm->reflectFunctions.idFromFieldObject(_currentThread, NULL, field);
+
+		I_32 result = (I_32)FALSE;
+		if ((NULL != fieldID)
```

Comment:
if `fieldId` is NULL there will be a pending exception, so instead of the NULL chec you can do an exception check `VM_VMHelpers::exceptionPending(_currentThread)` and then set the next target to `GOTO_THROW_CURRENT_EXCEPTION`

Reply:
addressed in e295c8